src/main/java/pl/uj/io/cuteanimals/model/action/PickupAction.java

```diff
@@ -0,0 +1,59 @@
+package pl.uj.io.cuteanimals.model.action;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import pl.uj.io.cuteanimals.model.GameState;
+import pl.uj.io.cuteanimals.model.Result;
+import pl.uj.io.cuteanimals.model.interfaces.IAction;
+import pl.uj.io.cuteanimals.model.interfaces.ICharacter;
+import pl.uj.io.cuteanimals.model.interfaces.IItem;
+import pl.uj.io.cuteanimals.model.interfaces.IResult;
+
+public class PickupAction implements IAction {
+    private final Map<String, IItem> items;
+    private List<String> args;
+
+    public PickupAction(Map<String, IItem> items) {
+        this.items = items;
+        this.args = new ArrayList<>();
+    }
+
+    @Override
+    public IResult execute(ICharacter character) {
+        if (!getAcceptableStates().contains(character.getCurrentGameState())) {
+            return new Result("This action cannot be executed now");
```

Comment:
Maybe change to sth like this:
```suggestion
            return new Result("You can't pick items right now.");
```
Just to keep immersion.